src/biogeophys/IrrigationMod.F90

```diff
@@ -1583,25 +1663,34 @@ subroutine CalcIrrigationNeeded(this, bounds, num_exposedvegp, filter_exposedveg
     do fp = 1, num_exposedvegp
        p = filter_exposedvegp(fp)
        c = patch%column(p)
+	   
+	   
+	   this%irrig_nsteps_per_day(c) = ((this%irrig_duration_column(c) + (this%dtime - 1))/this%dtime)
+	   
 
        if (check_for_irrig_patch(p)) then
-
-          ! Convert units from mm to mm/sec
-          this%sfc_irrig_rate_patch(p) = deficit_volr_limited(c) / &
-               (this%dtime*this%irrig_nsteps_per_day)
-          this%irrig_rate_demand_patch(p) = deficit(c) / &
-               (this%dtime*this%irrig_nsteps_per_day)
+          ! if using irrigation streams, do not set these two variables
+          ! TODO: determine volr limited rate based on prescribed input rates rather than soil moisture based irrigation rates
+          ! if (.not. use_irrigation_streams) then
+             ! Convert units from mm to mm/sec
+		  this%sfc_irrig_rate_patch(p) = deficit_volr_limited(c) / &
+			   (this%dtime*this%irrig_nsteps_per_day(c))
+		  this%irrig_rate_demand_patch(p) = deficit(c) / &
+			   (this%dtime*this%irrig_nsteps_per_day(c))
+          ! else
+          !    this%sfc_irrig_rate_patch(p) = 0._r8
+          !    this%irrig_rate_demand_patch(p) = 0._r8
+          ! endif
```

Comment:
- [ ] Is the `if (.not. use_irrigation_streams)` and `else` stuff meant to be commented out?
- [ ] Can you elaborate on that TODO?